Assets/BowAndArrowImproved/Scripts/BallonsSpawner.cs

```diff
@@ -1,35 +1,64 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Serialization;
 using Random = UnityEngine.Random;
 
 public class BallonsSpawner : MonoBehaviour
 {
     [SerializeField] private List<Transform> spawnLocations;
     [SerializeField] private ParticleSystem.MinMaxCurve spawnInterval;
     [SerializeField] private Ballon ballonPrefab;
-    [SerializeField] private int numOfBallons;
+    [SerializeField] private int baseNumOfBallons;
+    private int _curWaveNumOfBallons;
+    private int _currentWave = 0;
     
     private List<Ballon> _spawnedBallons = new List<Ballon>();
 
     private float _spawnTimer;
+
+    public void OnWaveDidEnd()
+    {
+        DestroyBallons();
+    }
+    
+    public void OnWaveDidStart()
+    {
+        GenerateWave();
+    }
```

Comment:
instead of having these 1-line functions that do nothing, simply call the DestroyBalloons() and GenerateWave() functions directly from the unity event